### PR TITLE
feat: add Blazor WASM demo

### LINKED ISSUE #13: Support for Blazor WASM
Having support for Blazor WASM will be nice.

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

provide a sample project for Blazor WASM, resolves #13

## Testing

image

### CODE REVIEW

File: sample-wasm/Pages/Home.razor.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+      private async Task OnLoginButtonClickAsync(MouseEventArgs obj)
+        {
+            await UserManager.SignInRedirectAsync();
+        }
+
+        private async Task OnLogoutButtonClickAsync(MouseEventArgs obj)
+        {
+            await UserManager.SignOutRedirectAsync();
+        }
```
Comment (darcyYe) [inline sample-wasm/Pages/Home.razor.cs:49]: nit: indentation

File: sample-wasm/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+## Configuration
+
+Create an `appsettings.Development.json` (or `appsettings.json`) with the following structure:
+
+```jsonc
+{
+  // ...
+  "IdentityServer": {
+    "Authority": "https://<your-logto-endpoint>/oidc",
+    "ClientId": "<your-logto-app-id>",
```
Comment (darcyYe) [inline sample-wasm/README.md:26]: No `clientSecret` needed?

File: sample-wasm/Pages/Home.razor.cs (hunk 1 above)
Comment (gao-sun) [reply sample-wasm/Pages/Home.razor.cs]: yeah fixed

File: sample-wasm/README.md (hunk 2 above)
Comment (gao-sun) [reply sample-wasm/README.md:26]: no it's SPA

**Review by darcyYe — APPROVED** [review]

**Review by simeng-li — APPROVED** [review]

File: sample-wasm/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+If you need to fetch an access token in JWT format for an API resource, add the following configuration:
+
+```jsonc
+{
+  // ...
+  "IdentityServer": {
+    // ...
+    "Resource": "https://<your-api-resource-indicator>",
+    "ExtraTokenParams": {
+      "resource": "https://<your-api-resource-indicator>" // Ensure the key is lowercase
```
Comment (charIeszhao) [inline sample-wasm/README.md:63]: Why are there duplicated "resource" configs?

Comment (gao-sun) [reply sample-wasm/README.md:63]: it's a long story. the underlying `oidc-client-ts` lib takes "Resource" for auth requests and takes "ExtraTokenParams" for token requests. i assume that was not designed for resource indicators.

**Pull request merged**